features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/interactiontonextview/InteractionToNextViewMetricResolver.kt

```diff
@@ -74,22 +72,36 @@ internal class InteractionToNextViewMetricResolver(
         return null
     }
 
-    private fun resolveLastInteraction(viewId: String, currentViewCreatedTimestamp: Long): InternalInteractionContext? {
-        val currentViewIdIndex = lastViewCreatedTimestamps.keys.indexOf(viewId)
-        val previousViewId = lastViewCreatedTimestamps.keys.elementAtOrNull(currentViewIdIndex - 1)
-        if (previousViewId != null) {
-            lastInteractions[previousViewId]?.let {
-                val context = PreviousViewLastInteractionContext(
-                    it.actionType,
-                    it.eventCreatedAtNanos,
-                    currentViewCreatedTimestamp
-                )
-                if (lastInteractionIdentifier.validate(context)) {
-                    return it
-                }
-            }
+    fun getState(viewId: String) = resolveMetric(viewId).let { metricValue ->
+        ViewInitializationMetricsState(
+            initializationTime = metricValue,
+            config = lastInteractionIdentifier.toConfig(),
+            noValueReason = if (metricValue == null) resolveNoValueReason(viewId) else null
+        )
+    }
+
+    @Suppress("ReturnCount")
+    private fun resolveNoValueReason(viewId: String): NoValueReason.InteractionToNextView {
+        // First of all, if there is no timestamp for the current view, all other metrics are meaningless.
+        val currentViewCreatedTimestamp = resolveCurrentViewCreationTimestamp(viewId)
+            ?: return NoValueReason.InteractionToNextView.UNKNOWN
+
+        // Second, if there is no previous view information, we are either on the very first screen of the app or
+        // lost the previous view information for some reason
+        val previousViewId = resolvePreviousViewId(viewId)
+            ?: return NoValueReason.InteractionToNextView.NO_PREVIOUS_VIEW
+
+        // Third - the case where there is no interaction from the previous view.
+        if (lastInteractions[previousViewId] == null) {
+            return NoValueReason.InteractionToNextView.NO_ACTION
         }
-        return null
+
+        // Finally - checking that the previous view interaction is eligible
+        resolveLastInteraction(previousViewId, currentViewCreatedTimestamp)
+            ?: return NoValueReason.InteractionToNextView.NO_ELIGIBLE_ACTION
```

Comment:
There is a corner case when view ID may change if session ID has changed https://github.com/DataDog/dd-sdk-android/blob/7387cc9494ff1321c969ad2664f6e36ff3dbeca1/features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt#L226-L229, but it is hard to address it anyway.